grafana/k6-studio
# feat: settings migration

## Description

This PR adds migrations to the settings schema, ensuring that changes to its structure remain backward-compatible. This allows the settings to work properly even when the schema is updated.

The same pattern will be applied to the Generator schema on a separate PR.



## How to Test



This PR contains one extra commit that is intended to be used for testing purposes and will be reverted before merging.

Before testing this branch, make sure your local settings file (`k6-studio-dev.json`) contains `usageReport.enabled` `false` and `appearance.theme` as `light` so you can validate the changes.

```json
"version": "1.0",
...
"usageReport": { "enabled": false },
"appearance": { "theme": "light" }
```

- Launch k6 studio from this branch
- Check that the theme and usage report are still the same
- Make changes to any other settings in the Settings dialog and check that the previous changes were migrated

Once the settings are saved, the file structure should look like this instead:

```json
"version": "1.0",
...
"general": {
   "usageReport": { "enabled": false },
   "appearance": { "theme": "light" }
}
```

## Checklist



## Screenshots (if appropriate):

## Related PR(s)/Issue(s)





Resolves https://github.com/grafana/k6-studio/issues/265
